giraffe/src/components/AnnotationLayer.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {FunctionComponent, useMemo} from 'react'
+import {AnnotationLayerConfig, AnnotationLayerSpec, LayerProps} from '../types'
+import {getAnnotationsPositions} from '../utils/annotationData'
+
+interface AnnotationLayerProps extends LayerProps {
+  spec: AnnotationLayerSpec
+  config: AnnotationLayerConfig
+}
+
+const ANNOTATION_OVERLAY_DEFAULT_STYLE = {
+  width: '100%',
+  height: '100%',
+}
+
+export const AnnotationLayer: FunctionComponent<AnnotationLayerProps> = props => {
+  const {height, width, spec, xScale, yScale} = props
+  const annotationsPositions = useMemo(
+    () => getAnnotationsPositions(spec.annotations, xScale, yScale),
+    [spec.annotations, xScale, yScale]
+  )
+
+  return (
+    <svg style={{...ANNOTATION_OVERLAY_DEFAULT_STYLE}}>
+      {annotationsPositions.map((annotation, i) =>
+        annotation.direction === 'y' ? (
+          <line
+            key={i}
+            x1="0"
+            x2={width}
+            y1={annotation.startValue}
+            y2={annotation.startValue}
+            stroke={annotation.color}
+          />
+        ) : (
+          <line
+            key={i}
+            x1={annotation.startValue}
+            x2={annotation.startValue}
+            y1="0"
+            y2={height}
+            stroke={annotation.color}
+          />
+        )
+      )}
+    </svg>
```

Comment:
if i'm understanding this correctly, the annotation spec contains a collection every annotation on the graph? is that right? and this component renders every annotation in the spec?

can there be multiple annotation specs or do all annotations follow the same one? should we separate the concept of an annotation spec from the annotations it specifies (in other words, move the `annotations` collection out of the spec)?

Reply:
Giraffe's config takes "layers". Our goal is to make these layers work seamlessly together, in other words, we can have multiple layers. Just like having multiple graph layers, we can have multiple annotation layers.

An `AnnotationLayerSpec` or annotation spec is just a blueprint for `PlotEnv` to handle annotation layers.

So, yes, ideally one annotation layer is enough to handle all annotations on the graph. But again, we want Giraffe to be multi-layer. Therefore, annotations can be used as multiple layers.

If that's not the case, then it's a bug.

Reply:
>Just like having multiple graph layers, we can have multiple annotation layers.

Is this what we want? We've discussed having annotations just be another layer in the graph, but I don't remember discussing having multiple annotations layers as a separate concept. I want to make sure that the design decisions we make can serve us well in the future

Reply:
What i meant here was that it's flexible enough to take multiple annotations layers. Not necessarily that we encourage the use of multiple annotations layers. If we didn't want multiple layers for annotations, then we simply don't pass in multiple layers. I would not go so far as to explicitly restrict annotations to one layer. That seems like an arbitrary restriction. Unless there is some benefit to doing so that i am not seeing.